### ISSUE TITLE
query, fragment get lost when evaluating javascript: urls

### ISSUE THREAD

**Issue by Ms2ger** [servo/servo#8259]
The new document should read "true", not "blah" in the following test cases:

Manual:

``` html
<!DOCTYPE html>
<a href='javascript:"blah"?"true":"false"'>...</a>
```

Automated (doesn't work in Servo; iframe load event doesn't seem to be fired):

``` html
<!DOCTYPE html>
<script>
var loaded = false;
function w(m) { console.log(m) }
function load(e) {
  w("load")
  var doc = e.target.contentDocument;
  if (!loaded) {
    w(doc.querySelector("a").click())
    loaded = true;
  } else {
    w(doc.documentElement.textContent)
  }
}
</script>
<iframe onload=load(event) src='data:text/html,<a href=&apos;javascript:"blah"?"true":"false"</a>&apos;>...'></iframe>
```

Spec: https://html.spec.whatwg.org/multipage/#javascript-protocol
Code: `ScriptTask::load` in `components/script/script_task.rs`

**Comment by Ms2ger** [comment]
Thanks to @zcorpan, for fragment:
```html
<!DOCTYPE html>
<a href='javascript:1//#%0a?2:4'>...</a>
```

**Comment by ConnorGBrewster** [comment]
Fixed by #8359?

**Comment by nox** [comment]
AFAIK, yes.
